### PR TITLE
Fix polynomial tests

### PR DESCRIPTION
**Pull request by juliuskunze**

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks - good catch! It looks like this was an oversight in an automated replace when we changed the import convention from `np`/`onp` to `jnp`/`np`.

File: tests/polynomial_test.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def args_maker():
         [jnp.zeros(leading, p.dtype), p, jnp.zeros(trailing, p.dtype)]),
 
     # order may differ (jnp.sort doesn't deal with complex numbers)
```
Comment (jakevdp) [inline tests/polynomial_test.py:61]: I think this comment is no longer true... we could replace `np.sort` with `jnp.sort` below, though that's not entirely necessary.

**Comment by jakevdp** [comment]
I spot checked the rest of the changes from #2973; I don't see any other issues like this.

**Review by jakevdp — APPROVED** [review]

File: tests/polynomial_test.py (hunk 1 above)
Comment (JuliusKunze) [reply tests/polynomial_test.py]: Indeed! I improved it.

**Comment by jakevdp** [comment]
Thanks!

**Pull request merged**